Standardise the use of `Visible`

BieleckiLtd opened the pull request:
MudBlazor currently uses the `Visible` and `IsVisible` properties. This PR aims to standardise the use of `Visible`.

## Description
If this PR is approved, the v7 migration guide must also be updated, as this makes a breaking change:

**MudCarouselItem**: replace `IsVisible` with `Visible`
**ChartSeries**: replace `IsVisible` with `Visible`
**SvgLegend**: replace `IsVisible` with `Visible`
**MudDialog**: replace `IsVisible` with `Visible`
**MudDialog**: replace `IsVisibleChanged` with `VisibleChanged`
**MudMessageBox**: replace `IsVisible` with `Visible`
**MudMessageBox**: replace `IsVisibleChanged` with `VisibleChanged`
**MudTooltip**: replace `IsVisible` with `Visible`
**MudTooltip**: replace `IsVisibleChanged` with `VisibleChanged`

Linked issues:
Negative property names should be discouraged #6131
v7.0.0 Migration Guide #8447

Standardise the use of `IsEnabled` and `Enabled` #8764
Standardise the use of `ItemDisabled` #8887
Standardise the use of `Checked`, `CheckedChanged` and `Checkable` #8825
Standardise the use of `Visible` #8832
Standardise the use of `Selected` and `SelectedChanged` #8886
Standardise the use of `Expanded`, `Expandable`, `IsExpanded` and `IsExpandable` #8718
Standardise the use of `Active` #8888
Standardise the use of `Open` and `OpenChanged` #8891
Standardise the use of `Editable` #8892
Standardise the use of `Hidden` and `HiddenChanged` #8952

## How Has This Been Tested?
unit

## Type of Changes
<!-- What type of changes does your code introduce? Put an `x` in only one box that applies best: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

BieleckiLtd committed:
Replace IsVisible with Visible

BieleckiLtd committed:
Resolve merge conflicts

BieleckiLtd committed:
Remove duplicated prop

henon commented:
@BieleckiLtd I wanted to merge this but I noticed that we probably should always also add the eventcallback to the illegal param detector for the cases where people didn't use the `@bind-` syntax. I.e. `IsVisibleChanged`. You could also add the event callbacks for your other standardization PRs, i.e. `IsCheckedChanged` etc.

BieleckiLtd committed:
Merge branch 'dev' into fix/standardise-visible

BieleckiLtd committed:
Added IsVisibleChanged to DetectIllegalRazorParametersV7

BieleckiLtd commented:
Added `IsVisibleChanged` to `DetectIllegalRazorParametersV7` and updated migration guide:

**MudDialog**: replace `IsVisibleChanged` with `VisibleChanged`
**MudMessageBox**: replace `IsVisibleChanged` with `VisibleChanged`
**MudTooltip**: replace `IsVisibleChanged` with `VisibleChanged`

henon commented:
Thanks @BieleckiLtd . Do you think we need to standardize other things as well or should we call it a day for v7?

henon commented:
@BieleckiLtd I think we may want to leave `IsValid` as is in `IForm` and `MudForm` as `EditForm` and `FluentValidation` both use `IsValid` and if we change `MudForm` to `Valid` it would actually increase inconsistency instead of decreasing it.

BieleckiLtd commented:
I'd like to do IsSelected and IsActive over the weekend. I believe IsDisabled was changed to Enabled, but I'll double check.

henon commented:
Great, I am changing `IsTouched` to `Touched` as we also have `Touched` already.
